feat: adding seed to shuffle method

Antnee opened the pull request:
Please refer to the updated README.md for more information

Antnee committed:
feat: adding seed to shuffle method

Please refer to the updated README.md for more information

Antnee committed:
test: adding invalid seed exception test

Brunty reviewed (approved):
![](https://media3.giphy.com/media/l0HegcvKKEPEPdOE0/giphy.gif)

Antnee committed:
refactor: removing non-int shuffle seed support

I really didn't like the solution, so I'm removing it. I don't think
that it belongs inside the collection library, anyway

Antnee commented:
I've simplified the shuffle seed solution. I really didn't like that we were creating an integer from a slice of an MD5 hash... It just felt like the wrong place, so now you _have_ to pass in an integer value if you want to see the PRNG prior to shuffling. If you like this still, @Brunty, I'm happy to merge and tag
